worlds/ufo50/barbuta/rules.py

```diff
@@ -0,0 +1,86 @@
+    regions["Starting Area"].connect(regions["Blood Sword Room"])
+    # pin via G2, umbrella via H3
+    regions["Starting Area"].connect(regions["G3 and Nearby"],
+                                     rule=lambda state: state.has_any((pin, umbrella), player))
+    regions["Starting Area"].connect(regions["F7 and Nearby"],
+                                     rule=lambda state: state.has(pin, player))
+    regions["Starting Area"].connect(regions["Mimic Room"],
+                                     rule=lambda state: state.has(wand, player))
+    regions["Starting Area"].connect(regions["C7 above Ladders"],
+                                     rule=lambda state: state.has(candy, player))
+
+    regions["Platforms above D4"].connect(regions["Bat Altar"])  # drop down off the left side
+    regions["Platforms above D4"].connect(regions["Above Entrance"])  # drop down off right side, need to break blocks
+    regions["Platforms above D4"].connect(regions["Boss Area"],
+                                          rule=lambda state: state.has(key, player))
+
+    regions["Above Entrance"].connect(regions["Wand Trade Room"])  # walk through the fake wall after using the door
+
+    regions["G7 and Nearby"].connect(regions["Wand Trade Room"],
+                                     rule=lambda state: state.has(pin, player))
+
+    regions["Wand Trade Room"].connect(regions["G7 and Nearby"],
+                                       rule=lambda state: state.has(pin, player))
+
+    regions["C7 above Ladders"].connect(regions["Boss Area"],
+                                        rule=lambda state: state.has(broken_wall, player))
+
+    regions["Mimic Room"].connect(regions["Boss Area"])
+
+    set_rule(world.get_location("Barbuta - Chest - G2"),
+             rule=lambda state: state.has(pin, player))
+    set_rule(world.get_location("Barbuta - Egg Shop - B6"),
+             rule=lambda state: has_money(100, state, player))
+    set_rule(world.get_location("Barbuta - Upper Shop Candy - C1"),
+             rule=lambda state: has_money(100, state, player))
+    set_rule(world.get_location("Barbuta - Upper Shop Umbrella - C1"),
+             rule=lambda state: has_money(100, state, player))  # todo: verify price
+    set_rule(world.get_location("Barbuta - Lower Shop Umbrella - F2"),
+             rule=lambda state: has_money(100, state, player))
+    set_rule(world.get_location("Barbuta - Lower Shop Trash - F2"),
+             rule=lambda state: has_money(50, state, player))
+    set_rule(world.get_location("Barbuta - Lower Shop Pin - F2"),
+             rule=lambda state: has_money(200, state, player))
+    set_rule(world.get_location("Barbuta - Little Guy Breaks a Wall"),
+             rule=lambda state: has_money(500, state, player))
+    set_rule(world.get_location("Barbuta - Bat Altar"),  # todo: should we do the wand version too?
+             rule=lambda state: state.has(blood_sword, player))
```

Comment:
I don't think the blood sword is needed for the altar at all, I just went to double check and got it to spawn with both the regular sword and the wand.

I can confirm that having the wand makes the orb spawn in the blood sword chest though, and that this is the only special case in the chest code (technically it's in the item drop's creation code I guess? whatever).